engine/graphics/src/null/graphics_null.cpp

```diff
@@ -1356,12 +1363,22 @@ namespace dmGraphics
             tex->m_OriginalHeight = params.m_OriginalHeight;
         }
 
-        return StoreAssetInContainer(context->m_AssetHandleContainer, tex, ASSET_TYPE_TEXTURE);
+        ASSET_CONTAINER_LOCK_BEGIN(context);
+
+        HTexture handle = StoreAssetInContainer(context->m_AssetHandleContainer, tex, ASSET_TYPE_TEXTURE);
+
+        ASSET_CONTAINER_LOCK_END(context);
```

Comment:
I used this pattern instead of the scoped lock, since we only need the lock in specific tests, i.e we don't use the async texture loading by default when using the null graphics library.